app/services/UserServiceImpl.js

```diff
@@ -29,7 +29,30 @@ module.exports = function(userDao) {
     });
   };
 
+  const signin = function(userData) {
```

Comment:
Changed name of function and arguments to authenticateUser. Dao function name and code also changed to run steps 1-3 from your comment. Takes username or email as argument.